sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,1359 @@
+        and the number of iterations. This might be useful in case one wants
+        to examine or store the transformation found after each iteration.
+
+    store_opt_result : bool, optional (default=False)
+        If True, the :class:`scipy.optimize.OptimizeResult` object returned by
+        :meth:`minimize` of `scipy.optimize` will be stored as attribute
+        ``opt_result_``.
+
+    verbose : int, optional (default=0)
+        If 0, no progress messages will be printed.
+        If 1, progress messages will be printed to stdout.
+        If > 1, progress messages will be printed and the ``iprint``
+        parameter of :meth:`_minimize_lbfgsb` of `scipy.optimize` will be set
+        to ``verbose - 2``.
+
+    random_state : int or numpy.RandomState or None, optional (default=None)
+        A pseudo random number generator object or a seed for it if int.
+
+    n_jobs : int, optional (default=1)
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Doesn't affect :meth:`fit` method.
+
+    Attributes
+    ----------
+    transformation_ : array, shape (n_features_out, n_features)
+        The linear transformation learned during fitting.
+
+    n_neighbors_ : int
+        The provided ``n_neighbors`` is decreased if it is greater than or
+        equal to  min(number of elements in each class).
+
+    n_iter_ : int
+        Counts the number of iterations performed by the optimizer.
+
+    opt_result_ : scipy.optimize.OptimizeResult (optional)
+        A dictionary of information representing the optimization result.
+        This is stored only if ``store_opt_result`` is True.
+
+    Examples
+    --------
+    >>> from sklearn.neighbors import LargeMarginNearestNeighbor
+    >>> from sklearn.neighbors import KNeighborsClassifier
+    >>> from sklearn.datasets import load_iris
+    >>> from sklearn.model_selection import train_test_split
+    >>> X, y = load_iris(return_X_y=True)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y,
+    ... stratify=y, test_size=0.7, random_state=42)
+    >>> lmnn = LargeMarginNearestNeighbor(n_neighbors=3, random_state=42)
+    >>> lmnn.fit(X_train, y_train) # doctest: +ELLIPSIS
+    LargeMarginNearestNeighbor(...)
+    >>> # Fit and evaluate a simple nearest neighbor classifier for comparison
+    >>> knn = KNeighborsClassifier(n_neighbors=3)
+    >>> knn.fit(X_train, y_train) # doctest: +ELLIPSIS
+    KNeighborsClassifier(...)
+    >>> print(knn.score(X_test, y_test))
+    0.933333333333
+    >>> # Now fit on the data transformed by the learned transformation
+    >>> knn.fit(lmnn.transform(X_train), y_train) # doctest: +ELLIPSIS
+    KNeighborsClassifier(...)
+    >>> print(knn.score(lmnn.transform(X_test), y_test))
+    0.971428571429
+
+    Notes
+    -----
+    Large margin nearest neighbor (LMNN) is a machine learning algorithm for
```

Comment:
This belongs in the user guide, not here.

Reply:
Done.